src/shogun/lib/SGStringList.cpp

```diff
@@ -93,12 +93,30 @@ SGStringList<T> SGStringList<T>::clone() const
 {
 	SGStringList<T> result(*this);
 
-	auto strings = SG_MALLOC(SGString<T>, num_strings);
+	auto strs = SG_MALLOC(SGString<T>, num_strings);
 
 	for (auto i : range(num_strings))
-		strings[i] = this->strings[i].clone();
+		strs[i] = this->strings[i].clone();
 
-	return SGStringList<T>(strings, num_strings, max_string_length);
+	return SGStringList<T>(strs, num_strings, max_string_length);
+}
+
+template <class T>
+bool SGStringList<T>::equals(const SGStringList<T>& other) const
+{
+	if (this->num_strings!=other.num_strings)
+		return false;
+
+	if (this->max_string_length!=other.max_string_length)
+		return false;
+
+	for (auto i : range(num_strings))
+	if (((SGString<T>)(this->strings[i])).equals(other.strings[i])==false)
```

Comment:
indentation problems. The CI build will give you a command to automagically fix this

Reply:
btw why is the cast to `SGString<T>` necessary?